youtube_dl/extractor/sdarot.py

```diff
@@ -0,0 +1,318 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..utils import (
+    sanitized_Request,
+    urlencode_postdata,
+)
+import re
+
+pragma = 'no-cache'
+origin = 'Origin: http://www.sdarot.pm'
+acceptEncoding = 'gzip, deflate'
+acceptLanguage = 'en-US,en;q=0.8,he;q=0.6'
+userAgent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36'
+contentType = 'application/x-www-form-urlencoded; charset=UTF-8'
+accept = '*/*'
+cacheControl = 'no-cache'
+xRequestedWith = 'XMLHttpRequest'
+connection = 'keep-alive'
+
+
+def gettoken(self, url, serie, season, episode):
+
+        video_id = "%s-%s-%s" % (serie, season, episode)
+
+        form1 = {
+            'preWatch': 'true',
+            'SID': serie,
+            'season': season,
+            'ep': episode
+        }
+
+        request1 = sanitized_Request('http://www.sdarot.pm/ajax/watch', urlencode_postdata(form1))
+        request1.add_header('Pragma', pragma)
+        request1.add_header('Origin', origin)
+        request1.add_header('Accept-Encoding', acceptEncoding)
+        request1.add_header('Accept-Language', acceptLanguage)
+        request1.add_header('User-Agent', userAgent)
+        request1.add_header('Content-Type', contentType)
+        request1.add_header('Accept', accept)
+        request1.add_header('Cache-Control', cacheControl)
+        request1.add_header('X-Requested-With', xRequestedWith)
+        request1.add_header('Connection', connection)
+        request1.add_header('Referer', url)
+
+        token = self._download_webpage(request1, video_id, None)
+
+        return token
+
+
+def getinfo(self, url, serie, season, episode, token):
+
+        video_id = "%s-%s-%s" % (serie, season, episode)
+
+        form2 = {
+            'watch': 'false',
+            'token': token,
+            'serie': serie,
+            'season': season,
+            'episode': episode
+        }
+
+        request2 = sanitized_Request('http://www.sdarot.pm/ajax/watch', urlencode_postdata(form2))
+        request2.add_header('Pragma', pragma)
+        request2.add_header('Origin', origin)
+        request2.add_header('Accept-Encoding', acceptEncoding)
+        request2.add_header('Accept-Language', acceptLanguage)
+        request2.add_header('User-Agent', userAgent)
+        request2.add_header('Content-Type', contentType)
+        request2.add_header('Accept', 'application/json, text/javascript, */*; q=0.01')
+        request2.add_header('Cache-Control', cacheControl)
+        request2.add_header('X-Requested-With', xRequestedWith)
+        request2.add_header('Connection', connection)
+        request2.add_header('Referer', url)
```

Comment:
Most likely default values should be enough. In any case any code duplication should be eliminated.